sqlitecluster/SQLite.cpp

```diff
@@ -205,43 +259,43 @@ string SQLite::read(const string& query) {
     return result[0][0];
 }
 
-// --------------------------------------------------------------------------
 bool SQLite::read(const string& query, SQResult& result) {
     // Execute the read-only query
     SASSERTWARN(!SContains(SToUpper(query), "INSERT "));
     SASSERTWARN(!SContains(SToUpper(query), "UPDATE "));
     SASSERTWARN(!SContains(SToUpper(query), "DELETE "));
     uint64_t before = STimeNow();
-    bool queryResult = SQuery(_db, "read only query", query, result);
+    bool queryResult = !SQuery(_db, "read only query", query, result);
     _readElapsed += STimeNow() - before;
     return queryResult;
 }
 
-// --------------------------------------------------------------------------
 bool SQLite::write(const string& query) {
-    SASSERT(!_readOnly);
     SASSERT(_insideTransaction);
     SASSERT(SEndsWith(query, ";"));                                         // Must finish everything with semicolon
     SASSERTWARN(SToUpper(query).find("CURRENT_TIMESTAMP") == string::npos); // Else will be replayed wrong
     // First, check our current state
     SQResult results;
-    SASSERT(SQuery(_db, "looking up schema version", "PRAGMA schema_version;", results));
+    uint64_t schemaBefore = 0;
+    uint64_t schemaAfter = 0;
```

Comment:
Out of curiosity, why do you prefer to define and initialize these up here, rather than defining and initializing them where they are used?  I'd say it might be an old C habit, but you're the most hard core C++ programmer I know. :)